### PR TITLE
Selects working

### PR DESCRIPTION
**Pull request by chasehensel**
@awans selects working!

### CODE REVIEW

**Comment by chasehensel** [comment]
@awans one thing I want to note is that the way I'm handling select / include as a separate query for delete doesn't really work for updates. As an example if you update a field in your where then when you rerun the query that value will no longer be included.

E.g

update("where {"name" : "andrew"}, data{"name" : "bob"}, include {"profile" : true}}

would fail if I changed the include to be a separate query at the end the way i am for delete

File: internal/api/operations/delete.go [hunk 1]
```diff
@@ -35,6 +35,17 @@ func (op DeleteOperation) Apply(tx db.RWTx) (*db.QueryResult, error) {
...
+
+	//final include
+	root = tx.Ref(op.ModelID)
+	clauses = handleFindMany(tx, root, op.FindArgs)
+	q = tx.Query(root, clauses...)
+	outs = q.All()
+	if len(outs) == 0 {
+		return nil, nil
+	}
+	out.ToOne = outs[0].ToOne
```
Comment (awans) [inline internal/api/operations/delete.go:47]: can you explain this one?

File: internal/api/operations/select.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+func handleSelects(tx db.Tx, parent db.ModelRef, s Select) []db.QueryClause {
+	clauses := []db.QueryClause{}
+	if s.Selecting {
+		var fields []string
+		for k, _ := range s.Fields {
+			fields = append(fields, k)
+		}
+		clauses = append(clauses, db.Select(parent, fields))
```
Comment (awans) [inline internal/api/operations/select.go:26]: looks reasonable to me!

File: internal/api/operations/update.go [hunk 3]
```diff
@@ -36,6 +36,15 @@ func (op UpdateOperation) Apply(tx db.RWTx) (*db.QueryResult, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	root = tx.Ref(op.ModelID)
+	clauses = handleFindMany(tx, root, op.FindArgs)
+	q = tx.Query(root, clauses...)
+	final := q.All()
+	if len(final) == 1 {
```
Comment (awans) [inline internal/api/operations/update.go:44]: same as before.. not sure i understand this..

File: internal/db/query_execution.go [hunk 4]
```diff
@@ -191,6 +203,7 @@ func (qb Q) performJoinOne(tx *holdTx, outer []*QueryResult, j JoinOperation) []
 				outer[i].Empty()
 			}
 		}
+		qb.projectFields(qb.Selections[j.To.AliasID], outer)
```
Comment (awans) [inline internal/db/query_execution.go:206]: the fact that we need to call projectFields in so many places suggests a refactor is required here, but it's fine to punt on that for now :)

File: internal/db/query_interface.go [hunk 5]
```diff
@@ -214,6 +244,25 @@ func Filter(ref ModelRef, m Matcher) QueryClause {
...
+				selection.fields[field] = void{}
+			}
+		} else {
+			qb.Selections[ref.AliasID] = Selection{true, make(set)}
+			for _, field := range fields {
+				qb.Selections[ref.AliasID].fields[field] = void{}
+			}
+
+		}
+		fmt.Println(qb.Selections[ref.AliasID].fields)
```
Comment (awans) [inline internal/db/query_interface.go:262]: cleanup?

File: internal/api/operations/delete.go (hunk 1 above)
Comment (chasehensel) [reply internal/api/operations/delete.go:47]: It was to give the original record with the new includes / selects. What we talked about in terms of desired behavior for how things return

File: internal/api/operations/update.go (hunk 3 above)
Comment (chasehensel) [reply internal/api/operations/update.go:44]: Ditto above

File: internal/db/query_execution.go (hunk 4 above)
Comment (chasehensel) [reply internal/db/query_execution.go:206]: :)

**Comment by chasehensel** [comment]
@awans i don't think texts are working for me this weekend -- long story of trying to try to switch to fi when i was abroad -- but i responded to your review

File: internal/api/operations/delete.go (hunk 1 above)
Comment (awans) [reply internal/api/operations/delete.go:47]: OK please add a comment explaining it

**Comment by chasehensel** [comment]
@awans updated. Let me know if it's good to submit!

**Pull request merged**